src/Numerics/ODESolvers/LowStorageRungeKutta3NMethod.jl

```diff
@@ -0,0 +1,273 @@
+ RKW (unused) provides RK weight (last row in Butcher's tableau).
+
+The 3-N storage formulation from Fyfe (1966) is applicable to any any 4-stage,
+fouth Order RK scheme. It is implemented here as:
+
+```math
+  t_j = t^n + rkC_j * \\Delta t
+  dQ_j = dQ^*_j + f(Q_j,t_j)
+  Q_{j+1} = Q_{j} + \\Delta t { rkB_{j,1} dQ_j + rkB_{j,2} dR_j }
+  dR_{j+1} = dR_j + rkA_{j+1,2} dQ_j
+  dQ^*_{j+1} = rkA_{j+1,1} dQ_j
+```
+
+The available concrete implementations are:
+
+  - [`LS3NRK44Classic`](@ref)
+  - [`LS3NRK33Heuns`](@ref)
+
+### References
+
+   @article {Fyfe1966,
+       title = {Economical Evaluation of Runge-Kutta Formulae},
+       author = {Fyfe, David J.},
+       journal = {Mathematics of Computation},
+       volume = {20},
+       pages = {392-398},
+       year = {1966}
+    }
+"""
+mutable struct LowStorageRungeKutta3N{T, RT, AT, Nstages} <: AbstractODESolver
+    "time step"
+    dt::RT
+    "time"
+    t::RT
+    "rhs function"
+    rhs!
+    "Storage for RHS during the LowStorageRungeKutta3N update"
+    dQ::AT
+    "Secondary Storage for RHS during the LowStorageRungeKutta3N update"
+    dR::AT
+    "low storage RK coefficient array A (rhs scaling)"
+    RKA::Array{RT, 2}
+    "low storage RK coefficient array B (rhs add in scaling)"
+    RKB::Array{RT, 2}
+    "low storage RK coefficient vector C (time scaling)"
+    RKC::Array{RT, 1}
+    "RK weight coefficient vector W (last row in Butcher's tableau)"
+    RKW::Array{RT, 1}
+
+    function LowStorageRungeKutta3N(
+        rhs!,
+        RKA,
+        RKB,
+        RKC,
+        RKW,
+        Q::AT;
+        dt = 0,
+        t0 = 0,
+    ) where {AT <: AbstractArray}
+
+        T = eltype(Q)
+        RT = real(T)
+
+        dQ = similar(Q)
+        dR = similar(Q)
+        fill!(dQ, 0)
+        fill!(dR, 0)
+
+        new{T, RT, AT, length(RKC)}(RT(dt), RT(t0), rhs!, dQ, dR, RKA, RKB, RKC, RKW)
+    end
+end
+
+"""
+    dostep!(Q, lsrk3n::LowStorageRungeKutta3N, p, time::Real,
+            [slow_δ, slow_rv_dQ, slow_scaling])
+
+Use the 3N low storage Runge--Kutta method `lsrk3n` to step `Q` forward in time
+from the current time `time` to final time `time + getdt(lsrk3n)`.
+
+If the optional parameter `slow_δ !== nothing` then `slow_rv_dQ * slow_δ` is
+added as an additionall ODE right-hand side source. If the optional parameter
+`slow_scaling !== nothing` then after the final stage update the scaling
+`slow_rv_dQ *= slow_scaling` is performed.
+"""
+function dostep!(
+    Q,
+    lsrk3n::LowStorageRungeKutta3N,
+    p,
+    time,
+    slow_δ = nothing,
+    slow_rv_dQ = nothing,
+    in_slow_scaling = nothing,
+)
+    dt = lsrk3n.dt
+
+    RKA, RKB, RKC = lsrk3n.RKA, lsrk3n.RKB, lsrk3n.RKC
+    rhs!, dQ, dR = lsrk3n.rhs!, lsrk3n.dQ, lsrk3n.dR
+
+    rv_Q = realview(Q)
+    rv_dQ = realview(dQ)
+    rv_dR = realview(dR)
+    groupsize = 256
+
+    rv_dR .= -0
+    for s in 1:length(RKC)
+        rhs!(dQ, Q, p, time + RKC[s] * dt, increment = true)
```

Comment:
`increment` is going to be depreciated, should this be switch? (@simonbyrne and @thomasgibson)

Reply:
it's still supported for now, we will probably get rid of it once we integrate TimeMachine.jl though